### PR TITLE
chore: Add reference tokens and replace legacy color tokens in style dictionary

### PR DESCRIPTION
**Pull request by jkuelz**
### Description

This change introduces reference tokens for primary, neutral, and status colors into our design token system. This is one of a series of updates to implement better out of the box global theming. We are keeping the legacy tokens for now so that we don't break our internal package builds which still rely on them in the meantime. Because we are keeping both tokens, this update also adds an eslint rule to tell developers to use the new tokens if the old ones are referenced.

Related links, issue #, if available: [quip: ac9YAbqsd7e6], CR-229486129

### How has this been tested?

Ran through dev pipeline: AwsUi-v3-jkuelz

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build-tools/eslint/index.js [hunk 1]
```diff
@@ -3,4 +3,5 @@
 module.exports.rules = {
   'ban-files': require('./ban-files'),
   'prefer-live-region': require('./prefer-live-region'),
+  'no-legacy-tokens': require('./no-legacy-tokens'),
```
Comment (just-boris) [inline build-tools/eslint/index.js:6]: Good idea to add a linter rule, but could you add it via a PR to another repository: https://github.com/cloudscape-design/build-tools/tree/main/src/eslint

And while adding this, it also needs some tests for positive and negative matches. There are some examples already

File: build-tools/eslint/no-legacy-tokens.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            matches.forEach(token => {
+              const colorFamily = token.match(/color(Grey|Blue|Green|Red|Yellow)/)?.[1];
+              const suggestion = REFERENCE_TOKEN_SUGGESTIONS[`color${colorFamily}`] || 'semantic reference tokens';
+
+              context.report({
+                node,
+                messageId: 'no-legacy-tokens',
+                data: { token, suggestion },
+              });
+            });
```
Comment (just-boris) [inline build-tools/eslint/no-legacy-tokens.js:40]: This code is repeated two times and can be reused, right?
